common-npm-packages/webdeployment-common/packageUtility.ts

```diff
@@ -67,7 +67,7 @@ export class Package {
 
     public async isMSBuildPackage(): Promise<boolean> {
```

Comment:
Can we refactor this method to add a bit more clarity?
```suggestion
    public async isMSBuildPackage(): Promise<boolean> {
  if (this._isMSBuildPackage !== undefined) return this._isMSBuildPackage;

  const shouldCheckFiles = this.getPackageType() !== PackageType.folder;
  this._isMSBuildPackage = shouldCheckFiles && await checkIfFilesExistsInZip(this._path, ["parameters.xml", "systeminfo.xml"]);

  tl.debug(`Is the package an msdeploy package : ${this._isMSBuildPackage}`);

  return this._isMSBuildPackage;
}
```